ios/Classes/TiBarcodeModule.m

```diff
@@ -0,0 +1,564 @@
+         withObject:@{
+           @"message" : [error localizedDescription] ?: @"Unknown error occurred."
+         }];
+
+    if (!keepOpen) {
+      [self closeScanner];
+    }
+  }
+
+  [[[[TiApp app] controller] topPresentedController] presentViewController:barcodeViewController
+                                                                  animated:animate
+                                                                completion:^{
+                                                                  [[barcodeViewController scanner] setTorchMode:selectedLEDMode ?: MTBTorchModeOff];
+                                                                }];
+}
+
+- (void)freezeCapture:(id)unused
+{
+  ENSURE_UI_THREAD(freezeCapture, unused);
+  [[barcodeViewController scanner] freezeCapture];
+}
+
+- (void)unfreezeCapture:(id)unused
+{
+  ENSURE_UI_THREAD(unfreezeCapture, unused);
+  [[barcodeViewController scanner] unfreezeCapture];
+}
+
+- (void)captureStillImage:(id)value
+{
+  ENSURE_UI_THREAD(captureStillImage, value);
+  ENSURE_SINGLE_ARG(value, KrollCallback);
+
+  [[barcodeViewController scanner] captureStillImage:^(UIImage *image, NSError *error) {
+    TiBlob *blob = [[TiBlob alloc] _initWithPageContext:[self pageContext]];
+    [blob setImage:image];
+    [blob setMimeType:@"image/png" type:TiBlobTypeImage];
+
+    NSDictionary *event = [NSDictionary dictionaryWithObject:blob forKey:@"image"];
+    [self _fireEventToListener:@"blob" withObject:event listener:(KrollCallback *)value thisObject:nil];
+  }];
+}
+
+- (void)cancel:(id)unused
+{
+  ENSURE_UI_THREAD(cancel, unused);
+
+  [self closeScanner];
+  [self fireEvent:@"cancel" withObject:nil];
+}
+
+- (void)setUseLED:(id)value
+{
+  ENSURE_TYPE(value, NSNumber);
+  [self replaceValue:value forKey:@"useLED" notification:NO];
+
+  selectedLEDMode = [TiUtils boolValue:value def:YES] ? MTBTorchModeOn : MTBTorchModeOff;
+
+  if (barcodeViewController) {
+    [[barcodeViewController scanner] setTorchMode:selectedLEDMode];
+  }
+}
+
+- (id)useLED
+{
+  return NUMBOOL(selectedLEDMode == MTBTorchModeOn);
+}
+
+- (void)setAllowRotation:(id)value
+{
+  DEPRECATED_REMOVED(@"Barcode.allowRotation", @"2.0.0", @"2.0.0");
+}
+
+- (void)setUseFrontCamera:(id)value
+{
+  ENSURE_TYPE(value, NSNumber);
+  [self replaceValue:value forKey:@"useFrontCamera" notification:NO];
+
+  selectedCamera = [TiUtils boolValue:value def:YES] ? MTBCameraFront : MTBCameraBack;
+  NSError *cameraError;
+
+  if (barcodeViewController) {
+    [[barcodeViewController scanner] setCamera:selectedCamera error:&cameraError];
+
+    if (cameraError) {
+      [self fireEvent:@"error"
+           withObject:@{
+             @"message" : [cameraError localizedDescription] ?: @"Unknown error occurred."
+           }];
+    }
+  }
+}
+
+- (id)useFrontCamera
+{
+  return NUMBOOL(selectedCamera == MTBCameraFront);
+}
+
+- (id)parse:(id)args
+{
+  ENSURE_SINGLE_ARG(args, NSDictionary);
+
+  id blob = [args valueForKey:@"image"];
```

Comment:
Try to prevent `id` whereever possible. Use `TiBlob` for this case and change all other id's if possible as well.